docs/API-EXAMPLE.md

```diff
@@ -369,7 +369,222 @@ main (int argc, char * argv[])
+
+```c
+#include <string.h>
+#include "jerry.h"
+
+/**
+ * Add param to 'this.x'
+ */
+static bool
+add_handler (const jerry_api_object_t *function_obj_p, /**< function object */
+             const jerry_api_value_t *this_p, /**< this arg */
+             jerry_api_value_t *ret_val_p, /**< return argument */
+             const jerry_api_value_t *args_p, /**< function arguments */
+             const jerry_api_length_t args_cnt) /**< number of function arguments */
+{
+  jerry_api_value_t x_val;
+
+  /* Get 'this.x' */
+  if (jerry_api_get_object_field_value (jerry_api_get_object_value (this_p),
+                                        (const jerry_api_char_t *) "x",
+                                        &x_val))
+  {
+    /* Convert Jerry API values to double */
+    double x = jerry_api_get_number_value (&x_val);
+    double d = jerry_api_get_number_value (args_p);
+
+    /* Add the parameter to 'x' */
+    jerry_api_value_t res_val = jerry_api_create_number_value (x + d);
+
+    /* Set the new value of 'this.x' */
+    jerry_api_set_object_field_value (jerry_api_get_object_value (this_p),
+                                      (const jerry_api_char_t *) "x",
+                                      &res_val);
+
+  }
+
+  return true;
+} /* add_handler */
+
+int
+main (int argc, char * argv[])
+{
+  jerry_completion_code_t status = JERRY_COMPLETION_CODE_OK;
+
+  /* Initialize engine */
+  jerry_init (JERRY_FLAG_EMPTY);
+
+  /* Create a JS object */
+  const jerry_api_char_t my_js_object[] = "MyObject = \
+  { x : 12, \
+    y : 'Value of x is ', \
+    foo: function () \
+    { \
+      return this.y + this.x; \
+    } \
+  }";
+
+  jerry_api_value_t my_js_obj_val;
+  /* Evaluate script */
+  status = jerry_api_eval (my_js_object,
+                           strlen ((const char *) my_js_object),
+                           false,
+                           false,
+                           &my_js_obj_val);
+
+  jerry_api_object_t *object_p = jerry_api_get_object_value (&my_js_obj_val);
+
+  /* Create a JS function object and wrap into a jerry value */
+  jerry_api_value_t object_value;
+  object_value.type = JERRY_API_DATA_TYPE_OBJECT;
+  object_value.u.v_object = jerry_api_create_external_function (add_handler);
+
+  /* Set the native function as a property of previously created MyObject */
+  jerry_api_set_object_field_value (object_p,
+                                    (const jerry_api_char_t *) "add2x",
+                                    &object_value);
+  jerry_api_release_value (&object_value);
+
+  /* Free JavaScript value, returned by eval (my_js_object) */
+  jerry_api_release_value (&my_js_obj_val);
+
+  const jerry_api_char_t script[] = "var str = MyObject.foo (); \
+                                     print (str); \
+                                     MyObject.add2x (5); \
+                                     print (MyObject.foo());";
```

Comment:
@akiss77, what would be the best alignment for the closing quotes? (I updated the example in the previous comment)